docs/user/model_setup.md

```diff
@@ -0,0 +1,137 @@
+
+```
+class ExampleTranslationModel(TranslationModel):
+```
+
+To implement the model interface, define the following functions:
+
+#### Load function
+
+Implement the load function to load all model artefacts from the model directory specified in `ASSET_URI`. The model artefacts loaded here can be used by the model during prediction time, i.e. executing run_model().
+Set the value self.ready as 'True' to indicate that loading has successfully executed.
+
+```
+    def load(self):
+        model_path = AssetResolver.resolve_path()
+        if not os.path.exists(model_path):
+            raise ValueError('Model not found')
+        self.model = pickle.load(os.path.join(model_path, 'model.pkl'))
+        self.ready = True
+```
+
+#### Run model function
+
+The run model function should contain the business logic to obtain a prediction from the loaded model.
+
+Input:  
+The input to the run model function is a dictionary. This dictionary has a key "instances" having values in a list containing AI function based subclass of APIInput values, for example TranslationInput.
+
+Output:  
+The output to the run model function is a dictionary. This dictionary has a key "predictions" having values in a list containing AI function based subclass of APIOutput values, for example TranslationOutput.  
+The output is expected to return the predictions from the model in the same order as the input instances were received.
+
+```
+    def run_model(self, api_input: Dict[str, List[TranslationInput]]) -> Dict[str, List[TranslationOutput]]:
+        src_text = self.parse_inputs(api_input["instances"])
+
+        translated = self.model.generate(
+            **self.tokenizer(
+                src_text, return_tensors="pt", padding=True
+            )
+        )
+
+        predictions = []
+        for t in translated:
+            data = self.tokenizer.decode(t, skip_special_tokens=True)
+            details = TextSegmentDetails(text=data)
+            output_dict = {
+                "data": data,
+                "details": details
+            }
+            translation_output = TranslationOutput(**output_dict)
+            predictions.append(translation_output)
+            predict_output = {"predictions": predictions}
+        return predict_output
+```
+
+
+### The system and Python requirements files
+
+The bash.sh file:  
+This file implementation should include installing any system dependencies using bash commands.
+
+The requirements.txt file:  
+Include all python packages that you need to run the model by extracting the requirements using the command below
+
+```
+pip freeze >> requirements.txt
+```
+
+Remove aixplain-models, CUDA, Torch and Tensorflow requirements from this file as their latest versions come pre-baked into the hosting server.
```

Comment:
What if specific versions of CUDA or torch are required for a particular model? If they are added, do we override them?

Reply:
@mikelam-us-aixplain can you please remove this line from the documentation?